metis-indexing/metis-indexing-client/src/main/java/eu/europeana/indexing/client/IndexingWorker.java

```diff
@@ -0,0 +1,45 @@
+package eu.europeana.indexing.client;
+
+import eu.europeana.indexing.client.IndexingClient;
+import eu.europeana.metis.exception.IndexingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IndexingWorker {
+	private static final Logger LOGGER = LoggerFactory.getLogger(IndexingWorker.class);	
+	
+	private final IndexingClient indexingClient;
+	
+	/**
+	 * Constructor.
+	 * 
+	 * @param indexingClient The indexing client
+	 */
+	public IndexingWorker(IndexingClient indexingClient) {		
+		this.indexingClient = indexingClient;
+	}
+	
+	 /**
+	   * Publishes a record
+	   * 
+	   * @param record The record to be published
+	   * @return true (success) or false (failure)
+	   * @throws IndexingException
+	   */
+	public boolean publish(String record) throws IndexingException {

```

Comment:
All this entire class does is call the indexing client. Maybe integrate this method with the indexing client (which is also quite simple) and then remove the class.

Reply:
Fixed.

Reply:
Can we then at least give them a constructor that needs no argument? Let's give them a very clean interface! That gives us more freedom to refactor in the future.